experimental/uefi/app/runner

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Thin shell script invoked as a cargo runner to run the compiled efi firmware
+# in QEMU. Detects if kvm is supported, and sets qemu flags based on that.
+# Instead of this single runner script it would be preferable to use a different
+# runner based on whether the kvm feature is set. However, cargo does not
+# currently allow this. Ref: https://github.com/rust-lang/cargo/issues/8170
+
+readonly TARGET=$1
+
+qemu_flags=(
+  '-nodefaults'
+  '-nographic'
+  '-bios' '/usr/share/OVMF/OVMF_CODE.fd'
+  '-serial' 'file:target/console.log'
+  '-serial' 'stdio'
+  '-machine' 'q35'
+  '-device' 'isa-debug-exit,iobase=0xf4,iosize=0x04'
+)
+
+# Use kvm if supported, as it is required for certain features (that are gated
+# behind the cargo `kvm` feature flag. Note that systems that support kvm still
+# need to enable the cargo `kvm` feature for this crate in order to include
+# code that requires kvm in the compiled firmware. Ideally this check would
+# check the cargo `kvm` flag itself. However, cargo does not expose this
+# information to the runner.
+# Ref: https://doc.rust-lang.org/cargo/reference/environment-variables.html
+if [[ -e "/dev/kvm" ]]; then
+  qemu_flags+=(
+    '-enable-kvm'
+    '-cpu' 'Broadwell-IBRS'
+  )
+fi
+
+qemu-system-x86_64 "${qemu_flags[@]}" -kernel "${TARGET}"
+
```

Comment:
Not a fan of this solution, but there does not currently seem to be a better way to do this. As long as we want this crate to run in GitHub CI we'll need to support hosts that do not support kvm.

Perhaps once we add a CI runner that supports virtualization we can remove this, only leaving the kvm reliant runner.